src/sources/kafka.rs

```diff
@@ -439,15 +504,69 @@ impl ReceivedMessage {
         }
     }
 
-    fn apply(&self, keys: &Keys<'_>, event: &mut Event) {
+    fn apply(&self, keys: &Keys<'_>, event: &mut Event, log_namespace: LogNamespace) {
         if let Event::Log(ref mut log) = event {
-            log.insert(keys.source_type, Bytes::from("kafka"));
-            log.insert(keys.timestamp, self.timestamp);
-            log.insert(keys.key_field, self.key.clone());
-            log.insert(keys.topic, Value::from(self.topic.clone()));
-            log.insert(keys.partition, Value::from(self.partition));
-            log.insert(keys.offset, Value::from(self.offset));
-            log.insert(keys.headers, Value::from(self.headers.clone()));
+            if let LogNamespace::Vector = log_namespace {
```

Comment:
:thinking: I don't think this needs to be an `if let` expression? Kind of confusing that its not just an 

```suggestion
            if log_namespace == LogNamespace::Vector {
```

Reply:
My thinking here is that it's more idiomatic to use pattern matching when the thing being compared is an enum.

Reply:
🤔 as a third choice, I think I'd usually `match` on the enum

Reply:
I went with the match since it's more explicit. Thanks!

Reply:
Oh haha! I didn't even notice there was an else clause there as I was tripped up by the if `let`.
In which case yes match is the way to go IMO as well.